pkg/security/probe/activity_dump_proto_v1.go

```diff
@@ -0,0 +1,268 @@
+	}
+
+	return ppi
+}
+
+func credentialsToProto(creds *model.Credentials) *adproto.Credentials {
+	if creds == nil {
+		return nil
+	}
+
+	pcreds := &adproto.Credentials{
+		Uid:            creds.UID,
+		Gid:            creds.GID,
+		User:           creds.User,
+		Group:          creds.Group,
+		EffectiveUid:   creds.EUID,
+		EffectiveGid:   creds.EGID,
+		EffectiveUser:  creds.EUser,
+		EffectiveGroup: creds.EGroup,
+		FsUid:          creds.FSUID,
+		FsGid:          creds.FSGID,
+		FsUser:         creds.FSUser,
+		FsGroup:        creds.FSGroup,
+		CapEffective:   creds.CapEffective,
+		CapPermitted:   creds.CapPermitted,
+	}
+
+	return pcreds
+}
+
+func fileEventToProto(fe *model.FileEvent) *adproto.FileInfo {
+	if fe == nil {
+		return nil
+	}
+
+	fi := adproto.FileInfoFromVTPool()
+	*fi = adproto.FileInfo{
+		Uid:          fe.UID,
+		User:         fe.User,
+		Gid:          fe.GID,
+		Group:        fe.Group,
+		Mode:         uint32(fe.Mode), // yeah sorry
+		Ctime:        fe.CTime,
+		Mtime:        fe.MTime,
+		MountId:      fe.MountID,
+		Inode:        fe.Inode,
+		InUpperLayer: fe.InUpperLayer,
+		Path:         fe.PathnameStr,
+		Basename:     fe.BasenameStr,
+		Filesystem:   fe.Filesystem,
+	}
+
+	return fi
+}
+
+func fileActivityNodeToProto(fan *FileActivityNode) *adproto.FileActivityNode {
+	if fan == nil {
+		return nil
+	}
+
+	pfan := adproto.FileActivityNodeFromVTPool()
+	*pfan = adproto.FileActivityNode{
+		Name:           fan.Name,
+		File:           fileEventToProto(fan.File),
+		GenerationType: string(fan.GenerationType),
+		FirstSeen:      timestamp(&fan.FirstSeen),
+		Open:           openNodeToProto(fan.Open),
+		Children:       make([]*adproto.FileActivityNode, 0, len(fan.Children)),
+	}
+
+	for _, child := range fan.Children {
+		pfan.Children = append(pfan.Children, fileActivityNodeToProto(child))
+	}
+
+	return pfan
+}
+
+func openNodeToProto(openNode *OpenNode) *adproto.OpenNode {
+	if openNode == nil {
+		return nil
+	}
+
+	pon := &adproto.OpenNode{
+		Retval: openNode.Retval,
+		Flags:  openNode.Flags,
+		Mode:   openNode.Mode,
+	}
+
+	return pon
+}
+
+func dnsNodeToProto(dn *DNSNode) *adproto.DNSNode {
+	if dn == nil {
+		return nil
+	}
+
+	pdn := &adproto.DNSNode{
+		Requests: make([]*adproto.DNSInfo, 0, len(dn.Requests)),
+	}
+
+	for _, req := range dn.Requests {
+		pdn.Requests = append(pdn.Requests, dnsEventToProto(&req))
+	}
+
+	return pdn
+}
+
+func dnsEventToProto(ev *model.DNSEvent) *adproto.DNSInfo {
+	if ev == nil {
+		return nil
+	}
+
+	return &adproto.DNSInfo{
+		Name:  ev.Name,
+		Type:  uint32(ev.Type),
+		Class: uint32(ev.Class),
+		Size:  uint32(ev.Size),
+		Count: uint32(ev.Count),
+	}
+}
+
+func socketNodeToProto(sn *SocketNode) *adproto.SocketNode {
+	if sn == nil {
+		return nil
+	}
+
+	return &adproto.SocketNode{
+		Family: sn.Family,
+		Bind: &adproto.BindNode{
+			Port: uint32(sn.Bind.Port),
+			Ip:   sn.Bind.IP,
+		},
+	}
+}
+
+func timestamp(t *time.Time) uint64 {
+	return uint64(t.Unix())
```

Comment:
UnixNano() ? Otherwise we're missing the nanosecond level precision :(

Reply:
What do you think we should do for wrong times ? For example currently `exit_time`s are returning `0001-01-01 00:00:00 +0000 UTC` -> `-6795364578871345152`

Reply:
Should be good now